### PR TITLE
Form

### PR DESCRIPTION
**Pull request by ANTOINE1128**
### Project requirements
- Implement the design from the sneak peek wireframes.
    - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

**Note: Analyze the wireframes deeply and try to identify all the necessary elements and how a user would navigate from one page to another**
    
All Users page wireframe Single User page wireframe All Posts by a given User page wireframe Single Post page wireframe

- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- **Hint:** In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,18 @@
 class ApplicationController < ActionController::Base
+  before_action :set_current_user
+
+  private
+
+  def set_current_user
+    user_id = params[:user_id]
+    @current_user = User.find_by(id: user_id)
+  end
```
Comment (tresorsawasawa) [inline app/controllers/application_controller.rb:9]: - [ ] As required for the project, kindly ensure that the `#set_current_user` method you created returns the first user from the database as now it is not the case. It should to something similar to this:
```ruby
def current_user
  User.first
end
```

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @like = Like.new(author_id: @user.id, post_id: @post.id)
+    if @like.save
+      flash[:notice] = 'Like was successfully created.'
+    else
+      flash[:alert] = 'Failed to add like.'
+    end
+    redirect_to user_post_path(@user, @post)
+  end
```
Comment (tresorsawasawa) [inline app/controllers/likes_controller.rb:14]: - [ ] As required for the project a user to be able to add a like to posts, kindly ensure that everything is working as expected. Right now, your application is showing this error while trying to add a new like.
image

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @ANTOINE1128 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- User can create a Post :heavy_check_mark: 
- User can create a Comment :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — COMMENTED** [review]
Hi @ANTOINE1128 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- You have submitted a closed pull request. Kindly reopen it or submit another pull request referencing the link to this pull request, so that the reviewer can be consistent with the changes required here. 👍 😎 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request closed without merging**